spec/message.abnf

```diff
@@ -83,5 +83,15 @@ content-char      = %x01-08        ; omit NULL (%x00), HTAB (%x09) and LF (%x0A)
 escaped-char = backslash ( backslash / "{" / "|" / "}" )
 backslash    = %x5C ; U+005C REVERSE SOLIDUS "\"
 
-; Whitespace
-s = 1*( SP / HTAB / CR / LF / %x3000 )
+; Optional whitespace
+owsp = *(s / bidi)
+
+; Required whitespace
+wsp = (owsp) 1*s (owsp)
+
+; Bidirectional marks and isolates
+; ALM / LRM / RLM / LRI, RLI, FSI & PDI
+bidi = %x061C / %x200E / %x200F / %x2066-2069
```

Comment:
Note that ALM is also permitted in `name-start` and `name-char`. We need to either drop it from here or there.